x/incentive/keeper/keeper.go

```diff
@@ -0,0 +1,203 @@
+package keeper
+
+import (
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/params/subspace"
+	"github.com/kava-labs/kava/x/incentive/types"
+)
+
+// Keeper keeper for the incentive module
+type Keeper struct {
+	key           sdk.StoreKey
+	cdc           *codec.Codec
+	paramSubspace subspace.Subspace
+	supplyKeeper  types.SupplyKeeper
+	cdpKeeper     types.CdpKeeper
+	accountKeeper types.AccountKeeper
+	codespace     sdk.CodespaceType
+}
+
+// NewKeeper creates a new keeper
+func NewKeeper(cdc *codec.Codec, key sdk.StoreKey, paramstore subspace.Subspace, sk types.SupplyKeeper, cdpk types.CdpKeeper, ak types.AccountKeeper, codespace sdk.CodespaceType) Keeper {
+
+	return Keeper{
+		key:           key,
+		cdc:           cdc,
+		paramSubspace: paramstore.WithKeyTable(types.ParamKeyTable()),
+		supplyKeeper:  sk,
+		cdpKeeper:     cdpk,
+		accountKeeper: ak,
+		codespace:     codespace,
+	}
+}
+
+func (k Keeper) Codespace() sdk.CodespaceType {
+	return k.codespace
+}
+
+// GetRewardPeriod returns the reward period from the store for the input denom and a boolean for if it was found
+func (k Keeper) GetRewardPeriod(ctx sdk.Context, denom string) (types.RewardPeriod, bool) {
+	store := prefix.NewStore(ctx.KVStore(k.key), types.RewardPeriodKeyPrefix)
+	bz := store.Get(types.GetDenomBytes(denom))
+	if bz == nil {
+		return types.RewardPeriod{}, false
+	}
+	var rp types.RewardPeriod
+	k.cdc.MustUnmarshalBinaryLengthPrefixed(bz, &rp)
```

Comment:
don't use `Must` for getters. Also, the standard on the SDK now is to use `BinaryBare` instead of `BinaryLengthPrefixed` since the former uses less bytes.

cc @karzak  ^

Reply:
It seems most(?) cosmos-sdk modules still use the Must pattern in getters (I checked evidence and upgrade as they are pretty new). If I switched to non-Must methods, would you recommend returning `(types.RewardPeriod, found, err)`?  Seems unnecessary as the structure of the store should make it impossible to insert items of the wrong type, no?

Reply:
👍 to using binary bare